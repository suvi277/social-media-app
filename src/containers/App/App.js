import React, { Fragment } from 'react';
import { Routes } from '@app/routes';
import { connect } from 'react-redux';
import { signOutUser } from '@app/actions';
import { Footer } from '@app/components/Footer';
import { Header } from '@app/components/Header';

const App = ({ isSignedIn, signOutUser }) => {
	return (
		<Fragment>
			<Header isAuthenticated={isSignedIn} signOutUser={signOutUser} />
			<main className="container">
				<Routes />
			</main>
			<Footer />
		</Fragment>
	);
};

const mapStateToProps = ({ userAccount }) => ({
	isSignedIn: userAccount.signedIn
});

const mapDispatchToProps = (dispatch) => ({
	signOutUser: () => dispatch(signOutUser())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
