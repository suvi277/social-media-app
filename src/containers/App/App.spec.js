import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import App from './App';
const mockStore = configureStore([]);

describe('App container', () => {
	let store, component;

	beforeEach(() => {
		store = mockStore({
			userAccount: {
				signedIn: false
			}
		});
		component = shallow(
			<Provider store={store}>
				<App />
			</Provider>
		);
	});
	it('should render with given state from Redux store', () => {
		expect(component).toMatchSnapshot();
	});
	it('should dispatch an action on button click', () => {});
});
