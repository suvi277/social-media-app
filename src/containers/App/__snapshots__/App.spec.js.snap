// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container should render with given state from Redux store 1`] = `ShallowWrapper {}`;
